src/diagnostics/internal/diagnosticsactionscontroller.cpp

```diff
@@ -52,3 +62,55 @@ void DiagnosticsActionsController::openUri(const mu::UriQuery& uri, bool isSingl
 
     interactive()->open(uri);
 }
+
+void DiagnosticsActionsController::saveDiagnosticFiles()
+{
+    if (configuration()->shouldWarnBeforeSavingDiagnosticFiles()) {
+        IInteractive::Result result = interactive()->warning(
+            trc("diagnostics", "Save diagnostic files?"),
+            trc("diagnostics", "This will create a .zip file with information about your MuseScore setup "
+                               "to help developers diagnose any problems you are having. "
+                               "You can inspect the contents of this file before sending it to anyone."),
+            { IInteractive::Button::Cancel, IInteractive::Button::Save }, IInteractive::Button::Save,
+            IInteractive::Option::WithIcon | IInteractive::Option::WithDontShowAgainCheckBox);
+
+        if (result.standardButton() != IInteractive::Button::Save) {
+            return;
+        }
+
+        configuration()->setShouldWarnBeforeSavingDiagnosticFiles(result.showAgain());
+    }
+
+    io::path_t path = interactive()->selectSavingFile(
+        qtrc("diagnostics", "Save diagnostic files"),
+        configuration()->diagnosticFilesDefaultSavingPath(),
+        { "(*.zip)" });
+
+    if (path.empty()) {
+        return;
+    }
+
+    qApp->setOverrideCursor(Qt::WaitCursor);
+    qApp->processEvents();
+
+    Ret ret = DiagnosticFilesWriter::writeDiagnosticFiles(path);
+
+    qApp->restoreOverrideCursor();
+
+    if (!ret) {
+        LOGE() << ret.toString();
+        return;
+    }
+
+    interactive()->revealInFileBrowser(path);
+}
```

Comment:
I would prefer that the implementation of this function was not in the controller. Maybe it's better to move it to DiagnosticFilesWriter (and make it a regular service with access through the interface)

Reply:
moved into a separate class